### PR TITLE
Fix logical expression

### PR DESCRIPTION
**Pull request by czpmango**
This PR modified LogicalExpression's behavior from the following three aspects：
 
```
 - Fix the logical expression(AND/OR/XOR) behavior to be consistent with Cypher when null expression is involved, refer to `https://s3.amazonaws.com/artifacts.opencypher.org/openCypher9.pdf` (search label : `Logical operations with null`).

 - Add BadNull expression check

 - Add empty expression check

 - Add empty check for Value::operator&/|/^

```

### CODE REVIEW

**Review by yixinglu — COMMENTED** [review]
Don't forget to fix `Value::operator&/|/^` comparator function

**Comment by czpmango** [comment]
> Don't forget to fix `Value::operator&/|/^` comparator function

Already fixed.

**Review by jievince — APPROVED** [review]
LGTM

File: src/common/expression/LogicalExpression.cpp [hunk 1]
```diff
@@ -24,62 +24,73 @@ const Value& LogicalExpression::eval(ExpressionContext& ctx) {
...
+        if (value.isBadNull()
+            || (value.isBool() && !value.getBool())) {
+            result_ = value;
+            return result_;
+        }
+        if (!value.isBool()) {
+            if (value.empty() && !result_.isNull()) {
+                result_ = value;
+            } else {
                 result_ = Value::kNullValue;
```
[the remark this answers is not in the document]
Comment (czpmango) [reply src/common/expression/LogicalExpression.cpp]: Already fixed.

**Review by CPWstatic — APPROVED** [review]
LGTM.

**Review by yixinglu — APPROVED** [review]

**Pull request merged**